### PR TITLE
[NPU] Support split `lm_head` for Qwen2

### PR DESCRIPTION
**Pull request by Oscilloscope98**
## Description

https://github.com/analytics-zoo/nano/issues/1763#issuecomment-2514266370

- Support split `lm_head` for Qwen2 for Python (cpp backend)
- Fit with Python (acc lib/L0 backend)
- Fit with C++ API
- Removed default `mixed_precision=True` in all-in-one and related examples

TODO:
- [x] Verify on Qwen2-7B (Python cpp backend)
- [x] Verify on MiniCPM-V-2_6 (Python acc lib backend)
- [x] Verify `mixed_precision=True`
- [x] Verify GW

### CODE REVIEW

**Review by plusbang — COMMENTED** [review]
Other LGTM

**Review by plusbang — APPROVED** [review]

**Review by rnwang04 — APPROVED** [review]
LGTM

**Review by cyita — APPROVED** [review]

File: python/llm/src/ipex_llm/transformers/npu_pipeline_model/qwen.py [hunk 1]
```diff
@@ -23,13 +23,14 @@
 
 
 def convert_lm_head_and_embedding(model, n_splits_linear, temp_dir, weight_dir,
```
Comment (cyita) [inline python/llm/src/ipex_llm/transformers/npu_pipeline_model/qwen.py:25]: Maybe remove `n_splits_linear`?

Comment (Oscilloscope98) [reply python/llm/src/ipex_llm/transformers/npu_pipeline_model/qwen.py:25]: Sure :)

**Pull request merged**